runtime/caml/misc.h

```diff
@@ -174,9 +178,15 @@ CAMLdeprecated_typedef(addr, char *);
 /* Prefetching */
 
 #ifdef CAML_INTERNALS
-#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
+#if (__has_builtin(__builtin_prefetch) || defined(__GNUC__)) && \
+    (defined(__i386__) || defined(__x86_64__) || \
+     defined(_M_IX86) || defined(_M_AMD64))
 #define caml_prefetch(p) __builtin_prefetch((p), 1, 3)
 /* 1 = intent to write; 3 = all cache levels */
+#elif defined(_M_IX86) || defined(_M_AMD64)
+#include <intrin.h>
```

Comment:
Have you checked with Godbolt that the Intel compiler (icc) provides `intrin.h`? This chunk should probably only get enabled if `configure` could detect `intrin.h` on the system.

Reply:
I've checked with ICC 2021.10 and 19.0.1 (probably not the most recent version) on Godbolt, and both define `__GNUC__`, with 2021.10 also defining `__has_builtin(__builtin_prefetch)`; so we're falling in the first case. I can guard the second case with `_MSC_VER` too, though.